### PR TITLE
Switch to gatling's java dsl

### LINKED ISSUE #19154: Switch to gatling's java dsl
##### **Overview of the feature request**

Gatling has been imho one of the best things we have in jhipster. While I still have https://github.com/jhipster/generator-jhipster/issues/13390 on my plate I would like to first migrate our gatling integration to the [(not so) new java dsl instead of the scala dsl](https://gatling.io/2021/11/gatling-3-7-java-dsl-kotlin-and-much-more/).

I guess this would also help to use the maven/gradle plugin again as we don't have the scala part anymore which was the reason why we have removed them. Having the plugins again and the java dsl will make it much easier for users to use, change and adapt the gatling tests (lot more now java, better ide(a) support, easier to start...)

##### **Motivation for or Use Case**

Make gatling much easier to use.

##### **Related issues or PR**

* https://github.com/jhipster/generator-jhipster/issues/13390

- [x] Checking this box is mandatory (this is just to show you read everything)

CC @mraible

### PR DESCRIPTION
**Pull request by murdos**
Fixes #19154

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mraible** [comment]
I like it! However, I'm not sure we have any CI tests that prove this works.

**Comment by atomfrede** [comment]
I guess we have never checked automatically if the Scala Tests still worked. But now it should be easier to have the maven/Gradle plugins for Gatling again

File: generators/server/templates/entity/src/test/java/gatling/simulations/EntityGatlingTest.java.ejs [hunk 1]
```diff
@@ -103,82 +118,84 @@ class <%= entityClass %>GatlingTest extends Simulation {
...
+        .check(status().is(302))
+        .check(header("Location").saveAs("loginUrl"))).exitHereIfFailed()
         .pause(2)
         .exec(http("Login Redirect")
         .get("${loginUrl}")
-        .silent
+        .silent()
         .headers(keycloakHeaders)
-        .check(css("#kc-form-login", "action").saveAs("kc-form-login"))).exitHereIfFailed
+        .check(css("#kc-form-login", "action").saveAs("kc-form-login"))).exitHereIfFailed()
```
Comment (atomfrede) [inline generators/server/templates/entity/src/test/java/gatling/simulations/EntityGatlingTest.java.ejs:131]: Not sure (it looks good according to the docs), but the failing tests look like a missing import

Comment (murdos) [reply generators/server/templates/entity/src/test/java/gatling/simulations/EntityGatlingTest.java.ejs:132]: Thanks for the hint, static import `io.gatling.javaapi.core.CoreDsl.css` was missing :-)

**Comment by atomfrede** [comment]
The failing tests seem unrelated. Looks like a compilation error with regards to kafka (I think we have a ticket for that) and an issue with mongo + reactive, which also has a ticket I think. CC @mshima @mraible I think both of you observed these issues during sb3 upgrade?

**Review by DanielFran — APPROVED** [review]

**Comment by murdos** [comment]
bounty claimed: https://opencollective.com/generator-jhipster/expenses/115505

**Comment by pascalgrimaud** [comment]
@murdos : approved

**Pull request merged**